components/CustomTable/CustomTable.tsx

```diff
@@ -0,0 +1,30 @@
+"use client";
+
+import React from "react";
+import { Table } from "antd";
+
+interface User {
+  [key: string]: any;
+}
+
+interface UserTableProps {
+  data: User[];
+}
+
+const CustomTable = ({ data }: UserTableProps) => {
+  const allKeys = Array.from(new Set(data.flatMap(Object.keys)));
+  const ConvertToNaturalString = (key: string) => {
+    return key
+      .replace(/([A-Z])/g, " $1")
+      .replace(/^./, (str) => str.toUpperCase());
+  };
+  const columns = allKeys.map((key) => ({
+    title: ConvertToNaturalString(key),
+    dataIndex: key,
+    key,
+  }));
+
+  return <Table columns={columns} dataSource={data} />;
+};
+
+export default CustomTable;
```

Comment:
Does your component contain an action column with buttons for editing and deleting records?

Reply:
I apologize, i will try to update it soon.